### ISSUE TITLE
[Accessibility] Errors in Lighthouse Report on Product Page

### ISSUE THREAD

**Issue by qasumitbagthariya** [woothemes/woocommerce#56239]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

While testing [PR #56119](https://github.com/woocommerce/woocommerce/pull/56119), I ran a Lighthouse accessibility audit on a product page and found errors. These issues need to be addressed to ensure compliance with accessibility standards.

### Expected behavior

The page should pass the Lighthouse Accessibility audit without errors.

### Actual behavior

Found accessibility-related errors in the Lighthouse report.

![Image](https://github.com/user-attachments/assets/2e30b7e9-060a-4a7f-a1c8-8757afd9ea97)

![Image](https://github.com/user-attachments/assets/1199089a-02d7-4c2b-8d00-414fefdcaa53)

### Steps to reproduce

1. Open a product page in WooCommerce.
2. Test on both Storefront and a WordPress block-based theme.
3. Open Dev Tools > Lighthouse.
4. Select the following options and analyze the page:
5. Navigation (Default)
6. Desktop
7. Accessibility

### WordPress Environment

WordPress: 6.7.2

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by jeffpaul** [comment]
@amberhinds this was flagged by QA, does this appear to be actionable from an accessibility perspective such that we should pick this up as well?

cc: @jamesckemp @vikrampm1

**Comment by amberhinds** [comment]
@jeffpaul - I'm not certain that either of these issues are created Woo.

The first screenshot is highlighting the navigation block.

![Image](https://github.com/user-attachments/assets/0a87d476-7f60-4c8d-b329-14691a00a522)

The heading order _might_ be Woo, but it depends on the theme.

@qasumitbagthariya can you provide more information about what theme you were testing in? Was it Twenty Twenty-Four or another theme?

Additionally, for the incorrect heading order, is that in the Related Products section or the Upsells section? Can you use the [HeadingsMap browser extension](https://chromewebstore.google.com/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi) on your testing page and provide a screenshot that shows all the headings on the page? 

I'm not certain that this is related to Woo. It could be caused by the theme or by how the theme is set up.

**Comment by gigitux** [comment]
Thanks for opening this issue!

> The heading order _might_ be Woo, but it depends on the theme.

In the case of a block theme with Woo default templates, the heading element is displayed with the Product Title block that it is a variation of the Post Title block:

https://github.com/woocommerce/woocommerce/blob/45b7639f809afaa467591164f724aab91c671a4c/plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/variations/elements/product-title.tsx#L20-L29


cc @kmanijak @imanish003

**Comment by qasumitbagthariya** [comment]
> [@jeffpaul](https://github.com/jeffpaul) - I'm not certain that either of these issues are created Woo.
> 
> The first screenshot is highlighting the navigation block.
> 
> ![Image](https://github.com/user-attachments/assets/0a87d476-7f60-4c8d-b329-14691a00a522)
> 
> The heading order _might_ be Woo, but it depends on the theme.
> 
> [@qasumitbagthariya](https://github.com/qasumitbagthariya) can you provide more information about what theme you were testing in? Was it Twenty Twenty-Four or another theme?
> 
> Additionally, for the incorrect heading order, is that in the Related Products section or the Upsells section? Can you use the [HeadingsMap browser extension](https://chromewebstore.google.com/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi) on your testing page and provide a screenshot that shows all the headings on the page?
> 
> I'm not certain that this is related to Woo. It could be caused by the theme or by how the theme is set up.


@amberhinds 

- During testing, I used the TT5 theme, version 1.1. While testing PR #56119, I ran a Lighthouse accessibility audit on a product page and found [errors](https://github.com/woocommerce/woocommerce/issues/56239).

 - I used the [HeadingsMap browser extension](https://chromewebstore.google.com/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi) to analyze the heading structure on my shop page. Please find the attached screenshot for reference.

![Image](https://github.com/user-attachments/assets/6217d643-1d73-4b77-9230-3d277b751263)

**Comment by amberhinds** [comment]
I was able to replicate this. On a fresh install, the Product Collection block on the shop page (which uses the Product Catalog FSE template) defaults to an H3, though it can be changed in the editor to an H2.

<img width="1728" alt="Image" src="https://github.com/user-attachments/assets/9d0e3ff0-b771-451e-877f-857d1d712078" />

Ideally, Woo should have accessible defaults and this should be addressed.

Here are the options:
- Change the product titles to an H2 and style them smaller in the block settings.
- Add an H2 above the Product Collection block.

**Comment by jeffpaul** [comment]
@jamesckemp @amberhinds is this something we should pick up as part of the a11y effort or is this something that can be left in the product backlog?

**Comment by amberhinds** [comment]
It should be included in the accessibility issue list.

**Comment by kmanijak** [comment]
[Here's](https://github.com/woocommerce/woocommerce/blob/cfe1431f61c36eec011e8911b7471b1a6806a69f/plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/constants.ts#L135) where the post-title is set to `level: 3` but I don't think there was a particular reason to do that except for sizing purposes.

Hence "Change the product titles to an H2 and style them smaller in the block settings" sounds like a way to go!

The same attributes are set in the blockified templates so the change should be propagated there as well. If possible, size should be preserved to avoid disruptions to existing stores using default templates defined: 

- plugins/woocommerce/templates/templates/blockified/archive-product.html
- plugins/woocommerce/templates/templates/blockified/product-search-results.html
- plugins/woocommerce/templates/templates/blockified/taxonomy-product_attribute.html
- plugins/woocommerce/templates/templates/blockified/taxonomy-product_cat.html
- plugins/woocommerce/templates/templates/blockified/taxonomy-product_tag.html
- plugins/woocommerce/templates/templates/blockified/taxonomy-product_brand.html

**Comment by Manussakis** [comment]
@qasumitbagthariya 

Woo does not create the issue related to the first screenshot in the description.


> Lists do not contain only `<li>` elements and script supporting elements (`<script>` and `<template>`).

The same issue can be verified in a fresh WP installation without WooCommerce installed.

![Image](https://github.com/user-attachments/assets/0d011707-a3e7-4576-ae55-39927015e075)

As for the second issue you flagged, I am working on it on the https://github.com/woocommerce/woocommerce/pull/56377 PR.

cc: @amberhinds
